relayer/relays/beacon/header/header.go

```diff
@@ -436,11 +436,9 @@ func (h *Header) findLatestCheckPoint(slot uint64) (state.FinalizedHeader, error
 	}
 	startIndex := uint64(lastIndex)
 	endIndex := uint64(0)
-	if uint64(lastIndex) > h.protocol.Settings.EpochsPerSyncCommitteePeriod {
-		endIndex = endIndex - h.protocol.Settings.EpochsPerSyncCommitteePeriod
-	}
 
 	syncCommitteePeriod := h.protocol.Settings.SlotsInEpoch * h.protocol.Settings.EpochsPerSyncCommitteePeriod
+	slotPeriodIndex := slot / syncCommitteePeriod
 
 	for index := startIndex; index >= endIndex; index-- {
```

Comment:
`endIndex` will also be `0`, is that correct?

Reply:
Yeah, that's primary change. _**By extending search range until the very beginning.**_

Basically what happened on Rococo setup is that we upgraded the beacon relayer a few weeks back, but we never upgraded the execution relayer so the bridge was down for about ~2-3 weeks.

So we need to search from the latest checkpoint backward until we find the matched checkpoint, previously we search for only 256 epochs at most which is not correct.